mevolib/fetch/BioSeqs.py

```diff
@@ -550,53 +464,48 @@ def update (self, email):
                     self.data = seq_dict
                     # Generate the corresponding report tuple
                     date_time = datetime.now().strftime('%Y/%m/%d %H:%M:%S')
                     self._report.append((date_time, 'entrez', entrez_db, query))
 
-    def write (self, bioseqs_file):
-        """
-        Save all sequences stored at the BioSeqs object in the 'bioseqs_file'
-        (in GENBANK format). A file with a detailed report of the sequences will
-        be created replacing the extension of 'bioseqs_file' by ".rep". If
-        'bioseqs_file' contains a relative path, the current working directory
-        will be used to get the absolute path. If any file already exists, it
-        will be overwritten without warning.
 
-        Arguments:
-            bioseqs_file  (string)
-                New BioSeqs sequence file.
+    def write (self, bioseqs_file: str):
+        """Save all sequences stored at the BioSeqs object in the 'bioseqs_file' (in GENBANK format). A file 
+        with a detailed report of the sequences will be created replacing the extension of 'bioseqs_file' 
+        by ".rep". If 'bioseqs_file' contains a relative path, the current working directory will be used to 
+        get the absolute path. If any file already exists, it will be overwritten without warning.
+
+        Args:
+            bioseqs_file: New BioSeqs sequence file.
 
         Raises:
-            IOError
-                If the path provided doesn't exist.
+            IOError: If the path provided doesn't exist.
         """
         data_filepath = get_abspath(bioseqs_file)
         report_filepath = os.path.splitext(data_filepath)[0] + '.rep'
         # Generate a single string with all the report content
         str_report = '\n'.join(['    '.join(x)  for x in self._report])
         # Write all the information in the BioSeqs files
         try:
-            SeqIO.write(viewvalues(self.data), data_filepath, 'genbank')
+            SeqIO.write(self.data.values(), data_filepath, 'genbank')
             with open(report_filepath, 'w') as report_file:
                 report_file.write('Num. sequences: {:d}\nHistory:\n' \
                                   '{:s}'.format(len(self), str_report))
         except IOError:
             raise
         except:
-            if (os.path.lexists(data_filepath)):
+            if os.path.lexists(data_filepath):
                 os.remove(data_filepath)
-            if (os.path.lexists(report_filepath)):
+            if os.path.lexists(report_filepath):
                 os.remove(report_filepath)
             raise
 
+
     def statistics (self):
-        """
-        Calculate the total number of sequences stored and the mean, standard
-        deviation, minimum and maximum values of their length.
+        """Calculate the total number of sequences stored and the mean, standard deviation, minimum and 
+        maximum values of their length.
```

Comment:
Missing `-> tuple` on type hint for `def`.
```suggestion
        """Returns the total number of sequences stored and the mean, standard deviation, minimum and 
        maximum values of their length.
```
`Returns:` section can be removed.